### PR TITLE
monitoring: high level gitserver metrics

### PR DESCRIPTION
**Pull request by uwedeportivo**
Composing two metrics (warning and alert) with values between zero (good) and one (really bad)

PR does this for gitserver

Grafana will display this as a Gauge with green for zero and red (yellow) for one

### CODE REVIEW

**Comment by uwedeportivo** [comment]
gitserverHealthWarning

File: cmd/gitserver/server/servermetrics.go [hunk 1]
```diff
@@ -43,8 +43,22 @@ func (s *Server) RegisterMetrics() {
 		Help:      "Amount of free space disk space on the repos mount.",
 	}, func() float64 {
 		var stat syscall.Statfs_t
+
```
Comment (slimsag) [inline cmd/gitserver/server/servermetrics.go]:
```suggestion
```

File: cmd/gitserver/server/servermetrics.go [hunk 2]
```diff
@@ -43,8 +43,22 @@ func (s *Server) RegisterMetrics() {
...
 		syscall.Statfs(s.ReposDir, &stat)
 		return float64(stat.Bavail * uint64(stat.Bsize))
 	})
 	prometheus.MustRegister(c)
+
+	c = prometheus.NewGaugeFunc(prometheus.GaugeOpts{
+		Namespace: "src",
+		Subsystem: "gitserver",
+		Name:      "disk_space_total",
+		Help:      "Amount of total disk space on the repos mount.",
```
Comment (slimsag) [inline cmd/gitserver/server/servermetrics.go]:
```suggestion
		Help:      "Amount of total disk space in the repos directory.",
```

File: cmd/gitserver/server/servermetrics.go [hunk 3]
```diff
@@ -43,8 +43,22 @@ func (s *Server) RegisterMetrics() {
...
+	}, func() float64 {
+		var stat syscall.Statfs_t
+
```
Comment (slimsag) [inline cmd/gitserver/server/servermetrics.go]:
```suggestion
```

File: cmd/gitserver/server/servermetrics.go [hunk 4]
```diff
@@ -43,8 +43,22 @@ func (s *Server) RegisterMetrics() {
 		Help:      "Amount of free space disk space on the repos mount.",
```
Comment (slimsag) [inline cmd/gitserver/server/servermetrics.go]:
```suggestion
		Help:      "Amount of free space disk space in the repos directory.",
```

**Comment by tsenart** [comment]
Is this ready for review given the WIP title?

**Comment by uwedeportivo** [comment]
@tsenart it's ready. i think @slimsag wants me to land it so he can iterate on it

**Comment by uwedeportivo** [comment]
The individual functions have range and domain in [0, 1].

metricFuncs

Combining them as a capped sum.

**Pull request merged**